ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0104__rhods_operator/0113__dsc_components.robot

```diff
@@ -301,28 +313,52 @@ Validate Feastoperator Removed State
 
 Validate Support For Configuration Of Controller Resources
     [Documentation]    Validate support for configuration of controller resources in component deployments
-    [Tags]    Operator    Tier1    ODS-2664      Integration
+    [Tags]    Operator    Tier1    ODS-2664      Integration  RHOAIENG-12811
     FOR   ${controller}    IN    @{CONTROLLERS_LIST}
-        ${rc}=    Run And Return Rc
-        ...    oc patch Deployment ${controller} -n ${APPLICATIONS_NAMESPACE} --type=json -p="[{'op': 'replace', 'path': '/spec/template/spec/containers/0/resources/limits/cpu', 'value': '600m'}]"    # robocop: disable
-        Should Be Equal As Integers    ${rc}    ${0}
-        ${rc}=    Run And Return Rc
-        ...    oc patch Deployment ${controller} -n ${APPLICATIONS_NAMESPACE} --type=json -p="[{'op': 'replace', 'path': '/spec/template/spec/containers/0/resources/limits/memory', 'value': '6Gi'}]"    # robocop: disable
-        Should Be Equal As Integers    ${rc}    ${0}
-        ${rc}=    Run And Return Rc
-        ...    oc patch Deployment ${controller} -n ${APPLICATIONS_NAMESPACE} --type=json -p="[{'op': 'replace', 'path': '/spec/template/spec/serviceAccountName', 'value': 'random-sa-name'}]"    # robocop: disable
-        Should Be Equal As Integers    ${rc}    ${0}
-
-        Wait Until Keyword Succeeds    3 min    0 sec
-        ...    Check Controller Conditions Are Accomplished      ${controller}
-
-        # Restore old values
-        # delete the Deployment resource for operator to recreate
-        ${rc}=    Run And Return Rc
-        ...    oc delete Deployment ${controller} -n ${APPLICATIONS_NAMESPACE}
-        Should Be Equal As Integers    ${rc}    ${0}
+        ${new_cpu_limit}=  Set Variable  1001m
+        ${new_memory_limit}=  Set Variable  4001Mi
+        ${new_image}=  Set Variable  registry.invalid/test:latest
+        ${new_replicas}  Set Variable  0
+
+        # overwrite some fields
+        Patch Controller Deployment  ${controller}  /spec/template/spec/containers/0/resources/limits/cpu  '${new_cpu_limit}'
+        Patch Controller Deployment  ${controller}  /spec/template/spec/containers/0/resources/limits/memory  '${new_memory_limit}'
+        Patch Controller Deployment  ${controller}  /spec/template/spec/containers/0/image  '${new_image}'
+        Patch Controller Deployment  ${controller}  /spec/replicas  ${new_replicas}
+
+        Sleep    10s  Give time for operator to potentially reconcile our changes
```

Comment:
Is this enough time? You have a 1 min 10 seconds wait for the reverted portion but only 10 seconds here.

Reply:
Good point, what I'm seeing is that the non-allowlisted values are reverted immediately, but it's true that's just the current implementation and it might change in the future (though probably unlikely). I originally also had 1 min wait here, the problem with that is that we iterate over all the controllers, so this test would take ~15 minutes.

One other option I considered is doing all the patches at once, then just a single wait, then verify if things got reverted. That makes the code somewhat ugly and less clear, but it's probably the safer approach, wdyt?